src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -496,6 +496,10 @@ def _forward(self, model_inputs, return_timestamps=False, generate_kwargs=None):
             else:
                 generate_kwargs["encoder_outputs"] = encoder(inputs, attention_mask=attention_mask)
 
+            # If the tokenizer has a pad token but the model doesn't, we add it to the `generate` call
+            if self.tokenizer.pad_token_id is not None and self.model.generation_config.pad_token_id is None:
```

Comment:
What's the reason for setting this here - this seems like something we should set on the model's generation config when creating the pipeline. 

If we do that, we can then directly check that the padding token is set as expected when the pipeline is created

Reply:
We can set it at __init__ time, yes. IMO not in generation config (permanently changes the object, = magic pattern unseen to the user), but storing it in `forward_params`. `forward_params` is then passed to `_forward`, which calls `generate`. We can even make it a generic pipeline operation, as opposed to pipeline-specific logic that needs to be added to each pipeline.

There is one case this alternative won't fix, though: if the user decides to set `tokenizer.pad_token_id` _after_ initializing the pipeline, only the current solution can handle it gracefully.

Happy to proceed with the change, or to keep as is. (Slight preference for the change, as it is solution with cleaner code)

Reply:
I like the forward params idea :) 

Since the pipeline is intended to be used as a one-liner way to get results, modifying the tokenizer seems counter to getting it to work. Is this something we commonly ask people to do for the pipelines? If so then probs best to leave as-is

Reply:
It's the first time I've seen this issue [tokenizer has a pad token but the model config doesn't], so it is not commonly asked. 

Going to push the change :)